fix: minor UI changes

sandesh-sp opened the pull request:
## Requirements
Change UI elements based on feedback from the team

## Changes
1. Change the Title to "OCO-3 Carbon Dioxide Snapshot Area Maps"
2. Add function to getMarkerColor based on constant MarkerColorMapping and fallback to hash based approach if no color found
3. Remove the text "normal" from configurable color map
4. Fix issues with timeline component - hover and active date state not syncing
5. Clear Map legend selection when clicked on "Home" button
6. Display 'N/A' for 0, undefined or null altitudes (todo: handle negative numbers)
7. Fix step size on configurable colormap having issues with floating point precision.

sandesh-sp committed:
Fix syncing of hover and active items between timeline and info card.

sandesh-sp committed:
Minor UI Fixes and marker color change

sandesh-sp committed:
Remove black outline/stroke in map marker SVG

sandesh-sp committed:
Change map marker colors

sandesh-sp committed:
Change default color map and unit

sandesh-sp committed:
Change title of vizItem ID from Visualization Item ID to Item ID

sandesh-sp committed:
Reset the map legend selection when clicked on home

sandesh-sp committed:
Refactor generateScale function to improve number formatting by accounting for floating point  precision

sandesh-sp committed:
Change text formatting for marker legend's category text

sandesh-sp committed:
Display N/A if sam type and altitude is null

sanzog03 commented on src/components/map/mapMarker/index.jsx:
```diff
@@ -112,9 +112,6 @@ const getMarkerSVG = (color, strokeColor = '#000000') => {
   return `
     <svg fill="${color}" width="30px" height="30px" viewBox="-51.2 -51.2 614.40 614.40" xmlns="http://www.w3.org/2000/svg">
       <g id="SVGRepo_bgCarrier" stroke-width="0"></g>
-      <g id="SVGRepo_tracerCarrier" stroke-linecap="round" stroke-linejoin="round" stroke="${strokeColor}" stroke-width="10.24">
```
Try to see if we can still put this by using a zero opacity color. Stroke could be useful in other usecases.

sanzog03 commented on src/components/ui/card/samInfoCard.tsx:
```diff
@@ -126,8 +126,12 @@ export const SamInfoCard = ({
+                <CaptionValue caption="SAM Type">
+                  {targetType ? targetType : 'N/A'}
+                </CaptionValue>
+                <CaptionValue caption="SAM Altitude">
```
@siddharth0248 does zero altitude for SAM makes sense? Should we put N/A for zero values as well?
@sandesh-sp change based on the reply 👆

sanzog03 reviewed (changes requested):
change these

sandesh-sp committed:
Add stroke option for map markers

sandesh-sp committed:
Added additional comment on timeline component

sandesh-sp committed:
Fix format of SAMs info card.
1. set Altitude to N/A if it's null, undefined or 0
2. Add unit 'm' after altitude
3. Capitalize first letter of SAM type

sanzog03 commented on src/components/ui/card/samInfoCard.tsx:
```diff
@@ -127,10 +128,10 @@ export const SamInfoCard = ({
+                  {targetType ? capitalizeFirstLetter(targetType) : 'N/A'}
                 </CaptionValue>
                 <CaptionValue caption="SAM Altitude">
-                  {targetAltitude != null ? targetAltitude : 'N/A'}
```
what if -ve values.
